### ISSUE TITLE
Assigning to a block variable changes the actual block (thus affecting `block_given?` and `yield`)

### ISSUE THREAD

**Issue by catsidhe** [mruby/mruby#6297]
```ruby
def foo1(&block)
  block = proc {}
  p block_given?
end
def foo2(&block)
  block = nil
  p block_given?
end
foo1
foo2 {}
```
Ruby prints `false` `true`, while mruby prints `true` `false`

**Comment by matz** [comment]
In CRuby, `block` is a copy of a given block reference. In mruby, `block` is a reference itself. This is primarily to reduce memory consumption. So please don't assign to the block arguments.

We will fix this someday. But the priority is low.

**Comment by catsidhe** [comment]
Interesting. Indeed it changes the block itself, so something like `block = nil; yield` would fail. I'll rephrase the issue title.

I don't mind this behavior if it's intentional, with a note added to `limitations.md` - or perhaps it would even be possible to issue a warning on such an assignment?

This alias/reference semantics is unique to block variables? Would it not be a potentially interesting new language feature, e.g.

```ruby
foo = 1
bar = ref(foo) # via some imaginary ref()
bar = 2 # changes foo
```

**Comment by matz** [comment]
To answer the question, it won't happen for other arguments, as far as I intended.

**Comment by matz** [comment]
This bug is fixed by 9c5812a4633e65b4c10cd212a3ae3f87a249a61c
